src/org/usfirst/frc/team694/robot/util/LimeLight.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Lime Light Client:
+ * Issues? Feel free to contact me as Sam Belliveau on Slack.
+ * the purpose of this project is to directly communicate with the Lime Light.
+ * This extends the included NetworkTableClient by taking table names from the
+ * Lime Light Documents. Using this will let you spend less time trouble
+ * shooting with Network Tables and spend more time making usable.
+ *
+ * Lime Light Docs: http://docs.limelightvision.io/en/latest/networktables_api.html#
+ */
+
+package org.usfirst.frc.team694.util;
+
+import org.usfirst.frc.team694.util.NetworkTableClient;
+
+public enum LED_MODE {
+  PIPELINE(0), FORCE_OFF(1), FORCE_BLINK(2), FORCE_ON(3);
+
+  private int codeValue;
+
+  LED_MODE(int value) { this.codeValue = value; }
+  public int getCodeValue() { return codeValue; }
+};
+
+public enum CAM_MODE {
+  VISION(0), DRIVER(1);
+
+  private int codeValue;
+
+  CAM_MODE(int value) { this.codeValue = value; }
+  public int getCodeValue() { return codeValue; }
+};
+
+public enum STREAM {
+  STANDARD(0), PIP_MAIN(1), PIP_SECONDARY(2);
+
+  private int codeValue;
+
+  STREAM_MODE(int value) { this.codeValue = value; }
+  public int getCodeValue() { return codeValue; }
+};
+
+public enum SNAPSHOT_MODE {
+  STOP(0), TAKE_TWO_PER_SECOND(1);
+
+  private int codeValue;
+
+  SNAPSHOT_MODE(int value) { this.codeValue = value; }
+  public int getCodeValue() { return codeValue; }
+};
+
+class LimeLight {
+  public static NetworkTableClient table = new NetworkTableClient("limelight");
+
+  /* “Best” Contour information */
+  // Whether the limelight has any valid targets (0 or 1)
+  public static boolean isValidTarget() {
+    return table.getDouble("tv") == 1;
+  }
+
+  // Horizontal Offset From Crosshair To Target (-27 degrees to 27 degrees)
+  public static final double MIN_X_OFFSET = -27;
+  public static final double MAX_X_OFFSET = 27;
+  public static double getTargetXOffset() {
+    return table.getDouble("tx");
+  }
+
+  // Vertical Offset From Crosshair To Target (-20.5 degrees to 20.5 degrees)
+  public static final double MIN_Y_OFFSET = -20.5;
+  public static final double MAX_Y_OFFSET = 20.5;
+  public static double getTY() {
+    return table.getDouble("ty");
+  }
+
+  // Target Area (0% of image to 100% of image)
+  public static final double MIN_TARGET_AREA = 0;
+  public static final double MAX_TARGET_AREA = 100;
```

Comment:
If you've decided on double for this type, makes more sense to go from 0 to 1

Reply:
Might want to wait and see, in the case that it is actually an int, which I cant tell because of the documentation, you might get weird rounding errors because of radix conversion. I don't know, more research is needed.

Reply:
You'll want to let folks know in the PR description that this hasn't been tested with a limelight yet, if it hasn't been

Reply:
sorry I'm new to git. I'm still kinda confused why this is a pull request. I thought it was a branch.